zip-0311.rst

```diff
@@ -0,0 +1,111 @@
+::
+
+  ZIP: 0311 
+  Title: FlyClient ZCash SPV
+  Champions: 
+  Status: Draft
+  Category: Consensus
+  Created: 2019-03-30
+  License: MIT 
+
+
+Terminology
+===========
+The key words "MUST", "SHOULD", and "MAY" in this document are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+Light client
+  A client that is not a full participant in the network of Zcash peers. It can send and receive payments, but does not store or validate a copy of the blockchain.
+
+High probability
+  An event occurs with high probability if it occurs with probability 1-O(1/2^λ), where λ is the security parameter.
+
+Negligible probability
+  An event occurs with negligible probability if it occurs with probability O(1/2^λ), where λ is the security parameter.
+
+Merkle mountain range (MMR)
+  A Merkle mountain range (MMR) is binary hash tree that allows for efficient appends of new leaves without changing the value of existing nodes [#PeterTodd]_.
+
+Hard fork 
+  A hard fork is a change to a protocol that makes transactions from older clients invalid. 
+  
+
+Abstract
+========
+A proposed implementation of the probabilistic verification FlyClient protocol [#FlyClient]_ on top of the existing ZCash reference light client protocol [#ZIPXXX]_. This can be implemented with a hard fork to include a MMR proof in each block header.
+
+Motivation
+==========
+FlyClient reduces the number of block headers needed for light client verification of a valid chain, from linear (as in the current reference protocol) to logarithmic in blockchain length. This verification is correct with high probability. It also allows creation of subchain proofs, so light clients need only check blocks later than the most recently verified block index. Following that, verification of a transaction inclusion within that block follows the usual reference protocol [#ZIPXXX]_. 
+
+A smaller proof size could facilitate ‘side-chaining’ with other blockchains such as Ethereum. It also reduces bandwidth and storage requirements for limited clients like mobile or IoT devices.
+
+
+Specification
+=============
+
+Probabilistic sampling
+```````````````````````
+Instead of downloading every block header, the FlyClient samples blocks from the full node with the given sampling distribution. Here, x is the relative aggregate weight at the given block (e.g. if x=1/2 at some block, it means that half the chain's difficulty has been amassed in the interval up to that block), and δ is the fraction of the blocks which are always checked by the client with probability 1.
+
+.. figure:: pdf.png
+    :align: center
+    :scale: 50 %
+    :figclass: align-center
```

Comment:
Because this ZIP uses additional files, all of its files should be under a `zip-0221` subdirectory.